From 4a1d23bc80701cc5fd85b963e90ce734249977b7 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Mon, 09 Jan 2023 10:31:55 +0200
Subject: [PATCH] [SW-0] pif_arc: clear die1 ready flag after flr reset

Due to a HW bug described in H9-5478, the die ready indication may
remain high while die1 is being reset.
As a workaround, firmware clears the die ready indication after
triggering the reset to the other die.

Change-Id: Ib016bb04795086aa343527b295fd29737486f1b9
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/app/pcie_pif/pcie_pif.c b/app/pcie_pif/pcie_pif.c
index e4c34c1..08e0f1e 100644
--- a/app/pcie_pif/pcie_pif.c
+++ b/app/pcie_pif/pcie_pif.c
@@ -181,6 +181,10 @@
 	regmap_writel(0, pif_arc.psoc2pif_flr_done);
 
 	set_flr_active();
+
+	/* clear die1 ready indication, WA for H9-5478 */
+	regmap_writel(0, pif_arc.psoc2pif_die_ready);
+
 	poll_flr_done();
 }
 
